resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -503,8 +511,9 @@ private[yarn] class YarnAllocator(
         allocatedContainerToHostMap.put(containerId, executorHostname)
       }
 
-      if (numExecutorsRunning < targetNumExecutors) {
+      if (numExecutorsRunning.get < targetNumExecutors) {
         if (launchContainers) {
+          numExecutorsStarting.incrementAndGet()
```

Comment:
note if we change the below back this needs to move outside if(launchContainers)